crypto/dilithium/pqcrystals_dilithium_ref_common/params.h

```diff
@@ -11,70 +15,42 @@
 #define Q 8380417
 #define D 13
 #define ROOT_OF_UNITY 1753
-
-#if DILITHIUM_MODE == 2
-#define K 4
-#define L 4
-#define ETA 2
-#define TAU 39
-#define BETA 78
-#define GAMMA1 (1 << 17)
-#define GAMMA2 ((Q-1)/88)
-#define OMEGA 80
-#define CTILDEBYTES 32
-
-#elif DILITHIUM_MODE == 3
-#define K 6
-#define L 5
-#define ETA 4
-#define TAU 49
-#define BETA 196
-#define GAMMA1 (1 << 19)
-#define GAMMA2 ((Q-1)/32)
-#define OMEGA 55
-#define CTILDEBYTES 48
-
-#elif DILITHIUM_MODE == 5
-#define K 8
-#define L 7
-#define ETA 2
-#define TAU 60
-#define BETA 120
-#define GAMMA1 (1 << 19)
-#define GAMMA2 ((Q-1)/32)
-#define OMEGA 75
-#define CTILDEBYTES 64
-
-#endif
-
 #define POLYT1_PACKEDBYTES  320
 #define POLYT0_PACKEDBYTES  416
-#define POLYVECH_PACKEDBYTES (OMEGA + K)
-
-#if GAMMA1 == (1 << 17)
-#define POLYZ_PACKEDBYTES   576
-#elif GAMMA1 == (1 << 19)
-#define POLYZ_PACKEDBYTES   640
-#endif
-
-#if GAMMA2 == (Q-1)/88
-#define POLYW1_PACKEDBYTES  192
-#elif GAMMA2 == (Q-1)/32
-#define POLYW1_PACKEDBYTES  128
-#endif
-
-#if ETA == 2
-#define POLYETA_PACKEDBYTES  96
-#elif ETA == 4
-#define POLYETA_PACKEDBYTES 128
-#endif
 
-#define CRYPTO_PUBLICKEYBYTES (SEEDBYTES + K*POLYT1_PACKEDBYTES)
-#define CRYPTO_SECRETKEYBYTES (2*SEEDBYTES \
-+ TRBYTES \
-+ L*POLYETA_PACKEDBYTES \
-+ K*POLYETA_PACKEDBYTES \
-+ K*POLYT0_PACKEDBYTES)
-#define CRYPTO_BYTES (CTILDEBYTES + L*POLYZ_PACKEDBYTES + POLYVECH_PACKEDBYTES)
+// Structure for ML-DSA parameters that depend on the parameter set.
+typedef struct {
+  size_t k;
+  size_t l;
+  size_t eta;
+  size_t tau;
+  size_t beta;
+  size_t gamma1;
+  int32_t gamma2;
+  size_t omega;
+  size_t c_tilde_bytes;
+  size_t poly_vech_packed_bytes;
+  size_t poly_z_packed_bytes;
+  size_t poly_w1_packed_bytes;
+  size_t poly_eta_packed_bytes;
+  size_t public_key_bytes;
+  size_t secret_key_bytes;
+  size_t bytes;
+} ml_dsa_params;
+
+// We define max values for some parameters because they are used
+// for static allocation.
+#define DILITHIUM_K_MAX (8)
+#define DILITHIUM_L_MAX (7)
+#define DILITHIUM_C_TILDE_BYTES_MAX (64)
+#define DILITHIUM_POLYW1_PACKEDBYTES_MAX (192)
+#define DILITHIUM_POLY_UNIFORM_ETA_NBLOCKS_MAX ((227 + STREAM256_BLOCKBYTES - 1)/STREAM256_BLOCKBYTES)
+#define DILITHIUM_POLYZ_PACKEDBYTES_MAX (576)
```

Comment:
These need asserts in ml_dsa_params_init to ensure these max constants are correct.